Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,559 @@
+                            chan_torrent.torrent_db = self.boosting_manager.torrent_db
+                            chan_torrent.channelcast_db = self.channelcast_db
+
+                            if chan_torrent.name:
+                                torrents.append(chan_torrent)
+
+                    return torrents
+
+                listtor = create_torrents(torrent_values, True,
+                                          {self.channel_id: self.channelcast_db.getChannel(self.channel_id)})
+
+                # dict {key_infohash(binary):Torrent(object-GUIDBTuple)}
+                self.unavail_torrent.update({t.infohash: t for t in listtor if t.infohash not in self.torrents})
+
+                # it's highly probable the checktor function is running at this time (if it's already running)
+                # if not running, start the checker
+                if not self.boosting_manager.is_pending_task_active(hexlify(self.source) + "_checktor"):
+                    self._logger.debug("Registering check torrent function")
+                    self.boosting_manager.register_task(hexlify(self.source) + "_checktor",
+                                                        LoopingCall(self._check_tor)).start(self.check_torrent_interval,
+                                                                                            now=True)
+
+    def _on_database_updated(self, dummy_subject, dummy_change_type, dummy_infohash):
+        self.database_updated = True
+
+    def get_source_text(self):
+        return self.channel.name if self.channel else None
+
+
+class RSSFeedSource(BoostingSource):
+    """
+    Credit mining source from a RSS feed.
+
+    # supported list (tested) :
+    # http://bt.etree.org/rss/bt_etree_org.rdf
+    # http://www.mininova.org/rss.xml
+    # https://kat.cr (via torcache)
+
+    # not supported (till now)
+    # https://eztv.ag/ezrss.xml (https link)
+    """
+
+    def __init__(self, session, rss_feed, boost_settings, callback):
+        BoostingSource.__init__(self, session, rss_feed, boost_settings, callback)
+
+        self.feed_handle = None
+
+        d = self._load_if_ready(rss_feed)
+        self.boosting_manager.register_task(str(self.source) + "_load", d, value=rss_feed)
+
+        self.torrent_store = self.session.lm.torrent_store
+        # keys : fake infohash, value : real infohash. Type : (length 20 string, binary)
+        self.fake_infohash_id = {}
+
+        self.title = ""
+        self.description = ""
+        self.total_torrents = 0
+
+    def _load(self, rss_feed):
+        self.feed_handle = self.session.lm.ltmgr.get_session().add_feed(
+            {'url': rss_feed, 'auto_download': False, 'auto_map_handles': False})
+
+        def wait_for_feed(d=None):
+            """
+            Wait until the RSS feed is no longer updating.
+            """
+            feed_status = self.feed_handle.get_feed_status()
+
+            if d is None:
+                d = defer.Deferred()
```

Comment:
You can also add this deferred to the taskManager to ensure it's cancelled upon stopping.